diff --git a/chromium/third_party/breakpad/breakpad/src/tools/linux/md2core/minidump-2-core.cc b/chromium/third_party/breakpad/breakpad/src/tools/linux/md2core/minidump-2-core.cc
index aade82c996d..fa7df048e5d 100644
--- a/chromium/third_party/breakpad/breakpad/src/tools/linux/md2core/minidump-2-core.cc
+++ b/chromium/third_party/breakpad/breakpad/src/tools/linux/md2core/minidump-2-core.cc
@@ -320,7 +320,7 @@ struct CrashedProcess {
     user_fpxregs_struct fpxregs;
 #endif
 #if defined(__aarch64__)
-    user_fpsimd_struct fpregs;
+    user_fpsimd_state fpregs;
 #endif
     uintptr_t stack_addr;
     const uint8_t* stack;
