Index: qtwebengine/configure.pri
===================================================================
--- qtwebengine.orig/configure.pri
+++ qtwebengine/configure.pri
@@ -438,8 +438,8 @@ defineTest(qtwebengine_isWindowsPlatform
         qtwebengine_platformError("requires MSVC or Clang (MSVC mode).")
         return(false)
     }
-    !qtwebengine_isMinWinSDKVersion(10, 19041): {
-        qtwebengine_platformError("requires a Windows SDK version 10.0.19041 or newer.")
+    !qtwebengine_isMinWinSDKVersion(10, 17763): {
+        qtwebengine_platformError("requires a Windows SDK version 10.0.17763 or newer.")
         return(false)
     }
     return(true)
